aslam_cv_common/cmake/detect_simd.cmake

```diff
@@ -0,0 +1,16 @@
+# Detect the preprocessor directives which are set by the compiler.
+execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dM -E -x c /dev/null
+                OUTPUT_VARIABLE PREPROCESSOR_DIRECTIVES)
+
+set(IS_SSE_ENABLED FALSE)
+set(IS_NEON_ENABLED FALSE)
+if (PREPROCESSOR_DIRECTIVES MATCHES "__SSE2__")
```

Comment:
Is `__SSE2__` actually covering SSE3 instructions as well? SSE3 does provide some additional instructions.

Reply:
@mfehr On current Intel/AMD processors, it seems to be quite likely that SSE3 is supported (see [here](https://stackoverflow.com/a/28186601)), but not strictly necessary. So I can add a check if it is really there.